### DISCUSSION TITLE
Love the integration! Quick question on connection

### DISCUSSION THREAD

**Discussion by nathanbutler** [sopelj/hass-ember-mug-component#40] [Q&A] [answered]
Thank you for making this its really awesome! Has anyone else noticed the constant connection/disconnection with their mugs? 

I am running the latest firmware on Ember Mug 2 with Home Assistant running in a docker container with Ubuntu as my host machine passing in that bluetooth. Everything with HA is working fine but find my Ubuntu machine is doing constant connection/disconnections from the mug. I believe it may be related to logic in the cup as it disconnects while stationary and reconnects when it moves. Makes the metrics a little annoying with mostly unavailable with brief moments of good data. But it could be just cup logic to save on battery life rather than constant Bluetooth connection. Just wanted to see if anyone else notices this?

**Comment by sopelj** [comment]
Thanks! I'm glad it can be of use to you. 

Is this a new issue that could be related to #38 or something that you experienced before that?

Before Home Assistant came out with their Bluetooth integration, this integration used to maintain a constant connection to the mug (unless it was out of range of course), but when it was migrated to the new one it had to be a little more flexible in case other devices need to use it. It should still maintain a connection as long as possible, but if you have other bluetooth devices or Home Assistant is doing things internally it might get disconnected. It should retry every 15 or 30 seconds to poll data though, unless it is unable to re-establish a connection.

If you want to try Bluetooth proxies, they are far more reliable and seem to have few issues.

**Reply by nathanbutler** [reply]
The bluetoothctl did help a lot with that initial connection. Yeah it could be that integration. I messed with some bluetooth settings in /etc/bluetooth/main.conf. Mainly FastConnectable=true and the reconnect to 5. It helped a little but still have it disconnecting. I will try the proxies and see if that helps! Thanks again I really appreciate it!

**Reply by sopelj** [reply]
I'm glad it helped. Is your device in proximity to your adapter? There seems to be a great many things that can cause interference with it. I think that is one of the reasons proxies are less problematic as they can be placed closer to the target and away from wires and adapters on the host.

The mug will also, as you pointed out, enter into sleep mode if it is left untouched and is not doing anything, and it will wake when moved. However, it should not be a problem when it is in use.

**Reply by nathanbutler** [reply]
Yeah that is the weird part. The cup is in direct line of sight maybe 10 feet away from the server that has builtin bluetooth. So I am not thinking it could be distant or object interference. I do have a lot of bluetooth devices in the same room though that could easily be just interference from devices like a laptop, tablet, headphones etc... But I am now pretty confident that its not this integration, my HA, or now it sounds like its not the cup either. So this is helping narrow it down. Or I guess it could be my cup in particular but not most cups as I saw a similar behavior with my ember app. I would have to re-pair every day to get my app to read my cup. Which is how I got turned on to this in the first place lol

**Reply by sopelj** [reply, marked as answer]
Oh ok, yeah, I definitely don't think the issue is range then. It might be interference though. The Home Assistant documentation lists of ton of [potential culprits](https://www.home-assistant.io/integrations/bluetooth/#bluetooth-interference-with-other-devices). Before Proxies were widely supported I ran a long USB 2 cable to move my dongle away from my server and that really helped in my case. I doubt your mug itself is a problem though. Yeah, the Ember app is pretty unreliable I had tons of issues with it and was one of the reasons I started working on this. haha.

**Reply by nathanbutler** [reply]
Awesome thanks for the help! Appreciate the time and effort. If I decide it's worth the bluetooth dongle proxy purchase route I will let you know how it works!

**Reply by sopelj** [reply]
Ok, thanks! Don't hesitate if you have any questions with it.
